test/engine-luatest/gh_4973_concurrent_alter_fails_test.lua

```diff
@@ -0,0 +1,53 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group('Replace during primary key altering')
+
+g.before_all(function()
+    g.server = server:new({})
+    g.server:start()
+end)
+
+g.after_all(function()
+    g.server:stop()
+    g.server = nil
+end)
+
+g.test_concurrent_alter = function()
+    g.server:exec(function()
+        local fiber = require('fiber')
+        local s = box.schema.space.create('test', {engine = 'memtx'})
+        s:create_index('pk',  {type='tree'})
+        local tuple_num = 2000
+        for i = 1, tuple_num do
+            s:replace{i, i, "old"}
+        end
+        local alter_in_process = false
+
+        local function disturb()
+            while not alter_in_process do fiber.sleep(0) end
```

Comment:
Do random `replaces`/`deletes` now (`replace` instead of `update` to insert tuple if it was previously deleted). Start updates before alter is over and do additional iteration when alter is over.